app/src/main/java/com/sdpteam/connectout/FirebaseActivity.java

```diff
@@ -0,0 +1,54 @@
+package com.sdpteam.connectout;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.concurrent.CompletableFuture;
+
+public class FirebaseActivity extends AppCompatActivity {
+
+    private final DatabaseReference db = FirebaseDatabase.getInstance().getReference();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_firebase);
+    }
+
+    public void get(View view) {
+        final CompletableFuture<String> future = new CompletableFuture<>();
+
+        db.child(getPhone()).get().addOnSuccessListener(data -> {
+            if (data.exists()) {
+                future.complete(data.getValue(String.class));
+            } else {
+                future.completeExceptionally(new NoSuchFieldException());
+            }
+        }).addOnFailureListener(future::completeExceptionally);
+
+        future.thenAccept(email -> {
+            final TextView field = findViewById(R.id.email);
+            field.setText(email);
+        });
+    }
```

Comment:
In my opinion, I think that this method should have a description as it isn't straightforward at first glance. This will make it much easier to use each other's code.